src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNINpHandle.cs

```diff
@@ -312,8 +322,29 @@ public override uint EnableSsl(uint options)
                 _validateCert = (options & TdsEnums.SNI_SSL_VALIDATE_CERTIFICATE) != 0;
                 try
                 {
-                    _sslStream.AuthenticateAsClient(_targetServer, null, SupportedProtocols, false);
-                    _sslOverTdsStream.FinishHandshake();
+                    if (_isTDS8)
+                    {
+#if NETCOREAPP
+                        SslApplicationProtocol TDS8 = new("tds/8.0");
```

Comment:
Nit: naming would be `tds8` since it's a local not a public member. I think I'd probably just call it `protocol` since tds is the value not the description of the variable that holds it.

Reply:
I would define a static list of client-supported protocols somewhere above (`private static clientApplicationProtocols s_Protocols = new List<SslApplicationProtocol>() { new SslApplicationProtocol("tds/8.0") }`) that you can reference on line 333 and remove this local variable completely.

EDIT: I now see that Wraith already suggested the same thing in SNITcpHandle.cs. We can use the same static reference here and there.

Reply:
can you check the new changes I made?

Reply:
in `SNIPhysicalHandle` create a new field `private static s_clientApplicationProtocols s_Protocols = new List<SslApplicationProtocol>(1) { new SslApplicationProtocol("tds/8.0") };` as David suggested and then in `SNINpHandle` and `SNITcpHandle` use it like this:
```
SslClientAuthenticationOptions sslClientOptions = new()
{
    TargetHost = _serverNameIndication,
    ApplicationProtocols = SNIPhysicalHandle.s_clientApplicationProtocols ,
    EnabledSslProtocols = SupportedProtocols,
    ClientCertificates = null,
};
```